ruma-events-macros/src/event_enum.rs

```diff
@@ -502,14 +502,18 @@ fn generate_accessor(
     }
 }
 
-fn field_return_type(name: &str) -> TokenStream {
+fn field_return_type(name: &str, var: &EventKindVariation) -> TokenStream {
     match name {
         "origin_server_ts" => quote! { ::std::time::SystemTime },
         "room_id" => quote! { ::ruma_identifiers::RoomId },
         "event_id" => quote! { ::ruma_identifiers::EventId },
         "sender" => quote! { ::ruma_identifiers::UserId },
         "state_key" => quote! { str },
-        "unsigned" => quote! { ::ruma_events::UnsignedData },
+        "unsigned" if &EventKindVariation::Sync == var => quote! { ::ruma_events::UnsignedSync },
+        "unsigned" if &EventKindVariation::RedactedSync == var => {
```

Comment:
These branches can be combined, either with an `||` or `[Variation::Sync, Variation::RedactedSync].contains` in the guard.